### DISCUSSION TITLE
How to save checkpoints more frequently than the end of every epoch?

### DISCUSSION THREAD

**Discussion by tareqalmuntasir7** [NVIDIA-NeMo/NeMo#3994] [Q&A] [answered]
Hi, 
Is there a way to save checkpoints more frequently? I am trying to make use of preemptible/spot cloud instances for cheaper training. But each epoch takes a few hours. I want to save progress more frequently so that I can resume from a later stage if my instance gets turned off. Can I save checkpoints depending on the number of steps or similar?. My goal is to resume from more recent weights. 
I didn't find anything in `ExpManger` configs except `every_n_epochs` parameter.

**Comment by titu1994** [comment, marked as answer]
Checkpoints are saved based on validation loss / metric. You can therefore perform validation between epochs to have more checkpoints. Use trainer.val_check_interval as a float < 1.0 (default 1.0 refers to 100% of a training epoch)

**Reply by muntasir2000** [reply]
Thanks a lot. Will try it out.
